packages/core/src/controllers/SiweController.ts

```diff
@@ -0,0 +1,87 @@
+import { subscribeKey as subKey } from 'valtio/utils'
+import { proxy, subscribe as sub } from 'valtio/vanilla'
+import type {
+  CreateSIWEMessageArgs,
+  SIWESession,
+  VerifySIWEMessageArgs
+} from '../utils/TypeUtil.js'
+
+import { SIWEStatus } from '../utils/TypeUtil.js'
+
+export interface SIWEControllerClient {
+  // -- Required --------------------------------------------- //
+  getNonce: () => Promise<string>
+  createMessage: (args: CreateSIWEMessageArgs) => string
+  verifyMessage: (args: VerifySIWEMessageArgs) => Promise<boolean>
+  getSession: () => Promise<SIWESession | null>
+  signOut: () => Promise<boolean>
+
+  // -- Optional --------------------------------------------- //
+
+  // Defaults true
+  enabled?: boolean
+  // In milliseconds, defaults to 5 minutes
+  nonceRefetchInterval?: number
+  // In milliseconds, defaults to 5 minutes
+  sessionRefetchInterval?: number
```

Comment:
`sessionRefetchIntervalMs` to signify milliseconds in the name itself (same for other similar bits)